src/v/cloud_storage/remote_segment.cc

```diff
@@ -1334,16 +1334,16 @@ remote_segment_batch_reader::read_some(
         if (
           _bytes_consumed != 0 && _bytes_consumed == new_bytes_consumed.value()
           && !_config.over_budget) {
-            auto context = fmt_with_ctx(
-              fmt::format,
+            vlog(
+              _ctxlog.error,
               "segment_reader is stuck, segment ntp: {}, _cur_rp_offset: {}, "
               "_bytes_consumed: "
               "{}",
               _seg->get_ntp(),
               _cur_rp_offset,
               _bytes_consumed);
-            throw stuck_reader_exception{
-              _cur_rp_offset, _bytes_consumed, context};
+            _is_unexpected_eof = true;
```

Comment:
Suggestion maybe for a later PR: perhaps we can expose the parser `_err` as a getter and check if we hit `end_of_stream` to confirm when setting this flag, so that future changes to the parser (perhaps if more benign error codes are added there) do not lead us into mistaking this condition for eof? 

We also have the advantage of being able to log the parser error which can help in debugging.

Reply:
you mean parser in the storage, right?

Reply:
yes